cli/cdsctl/config.go

```diff
@@ -98,3 +102,160 @@ func loadConfig(configFile string) (*cdsclient.Config, error) {
 
 	return conf, nil
 }
+
+func withAllCommandModifiers() []cli.CommandModifier {
+	return []cli.CommandModifier{cli.CommandWithExtraFlags, cli.CommandWithExtraAliases, withAutoConf()}
+}
+
+func withAutoConf() cli.CommandModifier {
+	return cli.CommandWithPreRun(
+		func(c *cli.Command, args *[]string) error {
+			if len(*args) >= len(c.Ctx)+len(c.Args) {
+				return nil
+			}
+
+			if _, err := repo.New("."); err != nil {
+				//Ignore error
+				return nil
+			}
+
+			if err := discoverConf(); err != nil {
+				return err
+			}
+
+			for _, arg := range c.Ctx {
+				if arg.Name == "project-key" {
```

Comment:
In a future PR I think it could be interesting to put these Kinds of value in const